src/In.ProjectEKA.HipService/Discovery/PatientDiscovery.cs

```diff
@@ -74,11 +90,38 @@ public class PatientDiscovery
                 return (null, error);
             }
 
+            Patient patient = patients.First(info => info.Identifier == patientEnquiryRepresentation.ReferenceNumber);
+
+            var toUnlinkedAccounts = AddToUnlinkedAccounts(patient.CareContexts, patient.Identifier);
+            if (!patient.CareContexts.Any() &&
+                !toUnlinkedAccounts)
+            {
+                return (null,
+                    new ErrorRepresentation(new Error(ErrorCode.FailedToGetLinkedCareContexts,
+                        "Unable to update unlinked care contexts")));
+            }
+
+            var careContexts = patientEnquiryRepresentation.CareContexts;
+            foreach (var careContext in careContexts)
+            {
+                careContext.ReferenceNumber = careContext.MaskedReferenceNumber;
+            }
+            
+            patientEnquiryRepresentation.CareContexts = careContexts;
             await discoveryRequestRepository.Add(new Model.DiscoveryRequest(request.TransactionId,
-                request.Patient.Id, patientEnquiryRepresentation.ReferenceNumber));
+                request.Patient.Id, patientEnquiryRepresentation.ReferenceNumber,
+                patient.MaskedIdentifier));
+
+            patientEnquiryRepresentation.ReferenceNumber = patient.MaskedIdentifier;
             return (new DiscoveryRepresentation(patientEnquiryRepresentation), null);
         }
 
+        private bool AddToUnlinkedAccounts(IEnumerable<CareContextRepresentation> careContexts,
+            string patientReferenceNumber)
+        {
+            return linkPatientRepository.SaveUnlinkedAccounts(careContexts, patientReferenceNumber).Result;
+        }

```

Comment:
What is this?

Reply:
Like the way we are storing masked patient reference number in the discovery request, we are storing the masked carecontext reference numbers of the patient in the table called UnlinkedAccounts

Reply:
Why do we need this? lets get on a call.